brave/brave-core
# Fix x-ipfs-path handling

Resolves https://github.com/brave/brave-browser/issues/25281 
For now if there is x-ipfs-header or 500-505 error code we check dnslink DNS record.
If there is such record, than current url is translated to ipns:// scheme and could be either used for redirect(if setting is enabled) or for showing the address bar button.
If there is only x-ipfs-path header then url is translated using content of the header. For example to http://localhost/ipfs/bafy.


Resolves 

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1) Enable autoredirect to dnslink in settings
2) Open en.wikipedia-on-ipfs.org. Check that site opens with ipns:// scheme.
3) Open https://unstoppabledomains.mypinata.cloud/ipfs/QmTZ4vvV926DN2HwZpd8djLyovFptqk3Xbct71YwJxt28A . Check that site opens as is and there is redirect button in address bar that leads to proper gateway url(depends on selected url in the settings).
4) Disable autoredirect to dnslink. Check that https://en.wikipedia-on-ipfs.org has address bar button which redirects to ipns://en.wikipedia-on-ipfs.org
5) Check that this buttons hides when navigate to non x-ipfs-path urls.
6) Check back and forward navigations between such urls.
